pkg/ccl/crosscluster/physical/standby_read_ts_poller_job.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package physical
+
+import (
+	"context"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/security/username"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/replication"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/errors"
+)
+
+const standbyReadTSPollInterval = 15 * time.Second
+
+type standbyReadTSPollerResumer struct {
+	job *jobs.Job
+}
+
+var _ jobs.Resumer = (*standbyReadTSPollerResumer)(nil)
+
+// Resume implements the jobs.Resumer interface.
+func (r *standbyReadTSPollerResumer) Resume(
+	ctx context.Context, execCtxI interface{},
+) (jobErr error) {
+	execCtx := execCtxI.(sql.JobExecContext)
+	return r.poll(ctx, execCtx.ExecCfg())
+}
+
+// OnFailOrCancel implements the jobs.Resumer interface.
+func (r *standbyReadTSPollerResumer) OnFailOrCancel(
+	ctx context.Context, _ interface{}, jobErr error,
+) error {
+	if jobs.HasErrJobCanceled(jobErr) {
+		err := errors.NewAssertionErrorWithWrappedErrf(
+			jobErr, "standby read ts poller job is not cancelable",
+		)
+		log.Errorf(ctx, "%v", err)
```

Comment:
we should throw an `errors.AssertionFailedf` here if the job was cancelled. It would suggest that something is deeply wrong with the job system